modules/darwin/src/main/ts/ephox/darwin/mouse/MouseSelection.ts

```diff
@@ -1,31 +1,28 @@
-import { Option } from '@ephox/katamari';
+import { Cell, Fun, Option } from '@ephox/katamari';
 import { Compare, SelectorFind, Element, EventArgs } from '@ephox/sugar';
 import * as CellSelection from '../selection/CellSelection';
 import { WindowBridge } from '../api/WindowBridge';
 import { SelectionAnnotation } from '../api/SelectionAnnotation';
 
-const findCell = function (target: Element, isRoot: (e: Element) => boolean) {
-  return SelectorFind.closest(target, 'td,th', isRoot);
-};
+const findCell = (target: Element, isRoot: (e: Element) => boolean) =>
+  SelectorFind.closest(target, 'td,th', isRoot);
 
-export default function (bridge: WindowBridge, container: Element, isRoot: (e: Element) => boolean, annotations: SelectionAnnotation) {
-  let cursor: Option<Element> = Option.none();
-  const clearState = function () {
-    cursor = Option.none();
-  };
+export default (bridge: WindowBridge, container: Element, isRoot: (e: Element) => boolean, annotations: SelectionAnnotation) => {
+  const cursor: Cell<Option<Element>> = Cell(Option.none());
```

Comment:
This would be better being a `Singleton` instead of a `Cell` and `Option`.

- [ ] Change to singleton

Reply:
Done